langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiChatModel.java

```diff
@@ -100,10 +108,11 @@ public AzureOpenAiChatModel(OpenAIClient client,
                                 List<AzureChatExtensionConfiguration> dataSources,
                                 AzureChatEnhancementConfiguration enhancements,
                                 Long seed,
-                                ChatCompletionsResponseFormat responseFormat,
+                                ResponseFormat responseFormat,
```

Comment:
I am wondering if we really need to set format in the constructor, as in the new `chat()` API it is possible to set it per request.

Reply:
I'm now doing a mix:
- You can set it up in the constructor, and it's the only way to set the "strict" boolean as otherwise it's not in the current `ChatRequest` object yet. I believe it's consistent with the other parameters, like `temperature`.
- Then the response format can be overridden in every request. I believe it's better as you can reuse your configuration for different requests (as a Spring user, I wouldn't like to have multiple instances of the same Spring bean here).

This will need to be refactored when `ChatRequest` supports the "strict" boolean, but that's shared with OpenAI and Gemini, so it should be in a different PR.